frontend/src/app/shared/dialog/objective-dialog/objective-form.component.ts

```diff
@@ -175,20 +179,22 @@ export class ObjectiveFormComponent implements OnInit, OnDestroy {
     this.unsubscribe$.complete();
   }
 
-  getSubmitFunction(id: number, objectiveDTO: any): Observable<Objective> {
-    if (this.data.action == 'duplicate') {
+  getSubmitFunction(id: number | undefined, objectiveDTO: any): Observable<Objective> {
```

Comment:
why can this suddenly by undefined

Reply:
it always was we just suppressed it using !